governance/pyth_staking_sdk/src/utils/vesting.ts

```diff
@@ -0,0 +1,58 @@
+import type { UnlockSchedule, VestingSchedule } from "../types";
+
+export const getUnlockSchedule = (options: {
+  pythTokenListTime: bigint;
+  vestingSchedule: VestingSchedule;
+}): UnlockSchedule => {
+  const { vestingSchedule, pythTokenListTime } = options;
+
+  if (vestingSchedule.fullyVested) {
+    return [];
+  } else if (vestingSchedule.periodicVestingAfterListing) {
+    return getPeriodicUnlockSchedule({
+      balance: vestingSchedule.periodicVestingAfterListing.initialBalance,
+      numPeriods: vestingSchedule.periodicVestingAfterListing.numPeriods,
+      periodDuration:
+        vestingSchedule.periodicVestingAfterListing.periodDuration,
+      startDate: pythTokenListTime,
+    });
+  } else if (vestingSchedule.periodicVesting) {
+    return getPeriodicUnlockSchedule({
+      balance: vestingSchedule.periodicVesting.initialBalance,
+      numPeriods: vestingSchedule.periodicVesting.numPeriods,
+      periodDuration: vestingSchedule.periodicVesting.periodDuration,
+      startDate: vestingSchedule.periodicVesting.startDate,
+    });
+  }
+  assertNever(vestingSchedule);
```

Comment:
Why is this needed?  Can the `vestingSchedule` be typed such that one of `fullyVested`, `periodicVestingAfterListing`, or `periodicVesting` is guaranteed to be set, or perhaps use an enum to represent the states so you can get exhaustivity checks from TS?

Reply:
fixed.